### PR TITLE
feat: Add ability to login with google

### PR DESCRIPTION
**Pull request by payers1**
To Dos
--
- [ ] Make sure we add the right redirect urls after we deploy
- [x] Invite team to google project so we all have access to client credentials

Mostly taken from here: https://github.com/vuejs/vue-router/tree/dev/examples/auth-flow

Docs here: https://developers.google.com/identity/sign-in/web/

### CODE REVIEW

File: src/auth.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+import Cookie from 'js-cookie'
+
 export default {
-  async authorize () {
-    isAuthorized = true
-    return Promise.resolve(isAuthorized)
+  async authorize (token) {
+    const inTwentyFourHours = new Date(new Date().getTime() + 24 * 60 * 60 * 1000);
+    Cookie.set('id_token', token, {expires: inTwentyFourHours})
+    return Promise.resolve(this.isAuthorized())
```
Comment (tstodter) [inline src/auth.js]: Why `Promise.resolve`?

Comment (payers1) [reply src/auth.js]: No reason, it can be taken out. But it was there and I wanted to have a small diff

File: src/components/Login.vue [hunk 2]
```diff
@@ -12,9 +12,25 @@ export default {
...
+    if (auth.isAuthorized()) {
+      this.$router.replace(this.$route.query.redirect || '/')
+    }
+    if (!window.gapi) {
+      return;
+    } else {
+      this.gapi = window.gapi;
+    }
+
+    gapi.signin2.render('g-signin2', {
```
Comment (tstodter) [inline src/components/Login.vue]: Does `gapi` resolve here?

Comment (tstodter) [reply src/components/Login.vue]: Like should it be `this.gapi`

Comment (tstodter) [reply src/components/Login.vue]: Guess it doesn't matter, since `window` is global scope. Scratch that.

Comment (payers1) [reply src/components/Login.vue]: `gapi` is a global that gets added to window when the `<script src="https://apis.google.com/js/platform.js" async defer></script>` loads

**Review by atomdmac — APPROVED** [review]
LGTM

File: src/auth.js (hunk 1 above)
Comment (atomdmac) [reply src/auth.js]: This was originally to test that the `async`/`await` stuff was working.

**Pull request merged**